### PR TITLE
Include advanced tutorials

### PR DESCRIPTION
**Pull request by benfitzpatrick**
Include advanced tutorial suites.

@oliver-sanders, please review.

### CODE REVIEW

File: suites/family-triggers/README [hunk 1]
```diff
@@ -0,0 +1,4 @@
+See http://metomi.github.io/rose/doc/rose-rug-advanced-tutorials-family-trigs.html.
+
+This suite lacks the suicide triggers explained in the later part of the tutorial,
+so tasks will fail and the suite will not automatically shutdown.
```
Comment (hjoliver) [inline suites/family-triggers/README]: Maybe rephrase this as it could be construed as saying tasks will fail _because_ the suicide triggers are missing.  "...so when tasks fail as expected, the suite will not automatically shutdown"?

File: suites/jinja2/suite.rc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            DETONATE:finish-all => stop
+        """
+[runtime]
+    [[root]]
+        [[[remote]]]
+            host = $ROSE_ORIG_HOST
+    [[DETONATE]]
+        script = detonate  # Diverge from the online tutorial - use a script instead.
+        [[[environment]]]
+            ROSE_TASK_APP = detonate
```
Comment (hjoliver) [inline suites/jinja2/suite.rc]: Remove - not using a Rose app?

File: suites/jinja2/suite.rc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{%- for num in range(ROCKET_NUMBER) %}
+{%- set num_label = '%02d' % num %}
+            ignite_rocket_{{ num_label }} => detonate_rocket_{{ num_label }}
+{%- endfor %}
+            DETONATE:finish-all => stop
+        """
+[runtime]
+    [[root]]
+        [[[remote]]]
+            host = $ROSE_ORIG_HOST
```
Comment (hjoliver) [inline suites/jinja2/suite.rc]: Is this necessary? (Isn't it the default?)  Else document?

File: suites/jinja2/bin/detonate [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -eu
+
+TERM=$(ls -drt -1 /dev/pts/* | tail -1); START=$((RANDOM%40 + 20))
+for i in {0..2}; do
+    printf "\r\033[2K %-"$((START-${#SOUND}/2))"s\033[1;3${COLOUR_CODE}m $SOUND \033[0m\r" " " > $TERM
+    SOUND=$(echo $SOUND | sed "s/\([^ ]\)/\1 /g")
+    sleep 0.15
+done
```
Comment (hjoliver) [inline suites/jinja2/bin/detonate:16]: This is brilliant, but maybe needs a comment on what's it's doing?  User's might think the seemingly magic result is a cylc feature. (... maybe we could send alerts to the terminal post-daemonization in this way!)

**Comment by hjoliver** [comment]
- 5.1 "complex recursions" - is "recursions" the right word here?  maybe "recurrence" or "sequence"?
- I think it would be worth writing down the the general recurrence forms like `Rn/START/INTERVAL` with a quick explanation of how "in context"  it reduces to things like `P1M` and `T00`.  Otherwise it looks like we support a list of ad hoc forms with not much rhyme or reason to unify them.
- Families: "these tasks can be grouped together for convenience" - this doesn't really capture the important reasons for families and inheritance: efficiency and maintainability via the ability to factor out and share all common configuration, and family triggering, and visualization.  I expect you might argue that's "too advanced" for a tutorial, but I think we should at least hint at the reasons, perhaps put in an appendix that is referenced in a `\note{...}` box.
- Jinja2: the example shows a simple loop over tasks; now need to mention that parameterized tasks are a cleaner way of doing this (and show the same with parameters)

**Comment by benfitzpatrick** [comment]
OK, now better.

**Comment by benfitzpatrick** [comment]
The only comment I haven't tried to address is parameterized tasks.

File: advanced-tutorials.tex [hunk 5]
```diff
@@ -93,7 +103,7 @@ \subsection{Families}
 \label{Families}
 
 Often a suite will contain a collection of similar tasks. With cylc these
-tasks can be grouped together for convenience.
+tasks can be grouped together for convenience. This can factor out a lot of repeated configuration to make suites more readable and maintainable.
```
Comment (hjoliver) [inline advanced-tutorials.tex]: This can _be used to_ factor out...?

**Comment by hjoliver** [comment]
Changes look good.

**Comment by benfitzpatrick** [comment]
Done.
